app/controllers/oai_controller.rb

```diff
@@ -134,6 +144,27 @@ def build_list_sets_response(organizations)
       end
     end.to_xml
   end
+
+  # See https://www.openarchives.org/OAI/openarchivesprotocol.html#Identify
+  def build_identify_response(earliest_date)
+    Nokogiri::XML::Builder.new do |xml|
+      xml.send :'OAI-PMH', oai_xmlns do
+        xml.responseDate Time.zone.now.iso8601
+        xml.request(oai_params.to_hash) do
+          xml.text oai_url
+        end
```

Comment:
i have an idea about how to refactor this out now that it's repeated in a few responses; will implement in next PR.

Reply:
Maybe something like this?

```
  def oai_template(xml)
      xml.send :'OAI-PMH', oai_xmlns do
        xml.responseDate Time.zone.now.iso8601
        xml.request(oai_params.to_hash) do
          xml.text oai_url
        end
        yield(xml)
      end
  end

  # See https://www.openarchives.org/OAI/openarchivesprotocol.html#Identify
  def build_identify_response(earliest_date)
    Nokogiri::XML::Builder.new do |xml|
     oai_template(xml) do
        xml.Identify do
          xml.repositoryName t('layouts.application.title')
          xml.baseURL oai_url
          xml.protocolVersion '2.0'
          xml.earliestDatestamp earliest_date.strftime('%F')
          xml.deletedRecord 'transient'
          xml.granularity 'YYYY-MM-DD'
          xml.adminEmail Settings.contact_email
        end
      end
    end.to_xml
  end
  # rubocop:enable Metrics/AbcSize
  # rubocop:enable Metrics/MethodLength
```

Reply:
I didn't read carefully enough...thought you wrote "have _no_ idea"

Reply:
This was the idea :)